pkg/cloud/azure/services/compute/virtualmachines.go

```diff
@@ -138,7 +141,34 @@ func (s *Service) MachineExists(machine *actuators.MachineScope) (bool, error) {
 	return vm != nil, nil
 }
 
+func getUserDataFromSecret(machine *actuators.MachineScope) (string, error) {
+	if machine.MachineConfig.UserDataSecret != nil {
+		userData := []byte{}
+		userDataSecret, err := machine.CoreClient.Secrets(machine.Namespace()).Get(machine.MachineConfig.UserDataSecret.Name, metav1.GetOptions{})
+		if err != nil {
+			return "", fmt.Errorf("failed to get userData secret %q", machine.MachineConfig.UserDataSecret.Name)
+		}
+		if data, exists := userDataSecret.Data["userData"]; exists {
+			userData = data
+		} else {
+			return "", fmt.Errorf("secret %q does not have field %q", machine.MachineConfig.UserDataSecret.Name, "userData")
+		}
+		encodedUserData := base64.StdEncoding.EncodeToString(userData)
```

Comment:
nit: drop the temporary, `return base64.StdEncoding.EncodeToString(userData), nil`.